chart/backup-restore-operator/index.vue

```diff
@@ -0,0 +1,199 @@
+import LabeledInput from '@/components/form/LabeledInput';
+import Banner from '@/components/Banner';
+import { get } from '@/utils/object';
+import { allHash } from '@/utils/promise';
+import { STORAGE_CLASS, SECRET, PV } from '@/config/types';
+import { mapGetters } from 'vuex';
+import { STORAGE } from '@/config/labels-annotations';
+
+export default {
+  components: {
+    Tab,
+    RadioGroup,
+    S3,
+    LabeledInput,
+    LabeledSelect,
+    Banner
+  },
+
+  hasTabs: true,
+
+  props: {
+    value: {
+      type:    Object,
+      default: () => {
+        return {};
+      }
+    },
+
+    mode: {
+      type:    String,
+      default: 'create'
+    }
+  },
+
+  async  fetch() {
+    const hash = await allHash({
+      storageClasses:    this.$store.dispatch('cluster/findAll', { type: STORAGE_CLASS }),
+      persistentVolumes: this.$store.dispatch('cluster/findAll', { type: PV }),
+      secrets:           this.$store.dispatch('cluster/findAll', { type: SECRET }),
+    });
+
+    this.secrets = hash.secrets;
+    this.storageClasses = hash.storageClasses;
+    this.persistentVolumes = hash.persistentVolumes;
+  },
+
+  data() {
+    const storageSource = this.mode === 'create' ? 's3' : this.getStorageSource(this.value);
+
+    return {
+      storageSource, secrets: [], storageClasses: [], persistentVolumes: []
+    };
+  },
+
+  computed: {
+    defaultStorageClass() {
+      return this.storageClasses.filter(sc => sc.metadata.annotations[STORAGE.DEFAULT_STORAGE_CLASS])[0];
+    },
+
+    storageClassNames() {
+      return this.storageClasses.reduce((total, each) => {
+        total.push(each.id);
+
+        return total;
+      }, []);
+    },
+
+    unboundPVs() {
+      return this.persistentVolumes.reduce((total, each) => {
+        if (each?.status?.phase !== 'bound') {
+          total.push(each.id);
+        }
+
+        return total;
+      }, []);
+    },
+
+    radioOptions() {
+      const options = ['s3', 'pickSC', 'pickPV'];
+      const labels = [
+        this.t('backupRestoreOperator.deployment.storage.options.s3'),
+        this.t('backupRestoreOperator.deployment.storage.options.pickSC'),
+        this.t('backupRestoreOperator.deployment.storage.options.pickPV'),
+      ];
+
+      if (this.defaultStorageClass) {
+        options.splice(1, 0, 'defaultSC');
+        labels.splice(1, 0, this.t('backupRestoreOperator.deployment.storage.options.defaultStorageClass', { name: this.defaultStorageClass.name }));
+      }
+
+      return { options, labels };
+    },
+
+    ...mapGetters({ t: 'i18n/t' })
+
+  },
+
+  watch: {
+    storageSource(neu) {
+      switch (neu) {
+      case 'defaultSC':
+        this.value.persistence.enabled = true;
+        this.value.s3.enabled = false;
+        this.value.persistence.storageClass = this.defaultStorageClass.id;
+        break;
+      case 'pickSC':
+        this.value.persistence.enabled = true;
+        delete this.value.persistence.volumeName;
```

Comment:
this should set storageClass in values.yaml to whichever storageClass user selects